langkit/documentation.py

```diff
@@ -374,17 +371,19 @@ def instantiate_templates(doc_dict):
         Return a hash for this unit, to be used in hash tables.
     """,
     'langkit.unit_reparse_file': """
```

Comment:
I'm actually wondering if sticky behavior in this one case wouldn't be good to keep: when you parsed from a file on disk before. Because, you're not directly passing a buffer, which was the original thing I warned about being dangerous in the context of the Python API, and in that case forgetting the old encoding seems weird and a bit counter intuitive.

No strong feeling either way though